contracts/cw-blotto/src/contract.rs

```diff
@@ -316,14 +318,24 @@ impl BlottoContract<'_> {
             // Sort army total stakes by staked amount, last item will be the winner
             army_totals.sort_by(|a, b| a.1.cmp(&b.1));
 
-            // TODO check for tie
+            // Check for tie on staked amounts
+            let mut unique_stakes = HashSet::new();
+            let is_tie = !army_totals
+                .iter()
+                .all(|(_, stake)| unique_stakes.insert(stake.to_string()));
```

Comment:
Nit: this will work for a battle with two armies, but there can be more than two. This will produce a false tie if there are more than two armies and two armies that aren't the winner have tied. As army_totals are sorted, we only care if the last two items are tied.